platform/web/retry/roundtripper.go

```diff
@@ -0,0 +1,66 @@
+package retry
+
+import (
+	"net/http"
+	"time"
+)
+
+// Config describes the configuration options for the middleware
+type Config struct {
+	MaxRetry int
+	Delay    time.Duration
+	ExceptOn []int
+}
+
+type retryRoundTripper struct {
+	roundTripper http.RoundTripper
+	retryConfig  Config
+}
+
+func (retryRoundTripper retryRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	doRequest := func(request *http.Request) (*http.Response, error) {
+		return retryRoundTripper.roundTripper.RoundTrip(request)
+	}
+
+	maxReTries := retryRoundTripper.retryConfig.MaxRetry
+
+	response, err := doRequest(req)
+	if err != nil || !isExceptStatus(response.StatusCode, retryRoundTripper.retryConfig.ExceptOn) {
+		for retries := 0; retries < maxReTries; retries++ {
+			time.Sleep(retryRoundTripper.retryConfig.Delay)
+
+			res, err := doRequest(req)
+			if retries == maxReTries-1 {
+				return res, err
+			}
+
+			if err != nil || !isExceptStatus(res.StatusCode, retryRoundTripper.retryConfig.ExceptOn) {
+				continue
+			}
+
+			return res, nil
+		}
+	}
+
+	return response, nil
+}
+
+func isExceptStatus(status int, exceptStatuses []int) bool {
+	for _, exceptStatus := range exceptStatuses {
+		if status == exceptStatus {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I am not sure I understand how generics would improve this function actually?

Reply:
it was more intended to be a joke because I am a bit annoyed by the amount of occurences of this kind of functions, but with generics you'd have a `find` function on slices that does that for you.